### DISCUSSION TITLE
Why was the tree view removed?

### DISCUSSION THREAD

**Discussion by johanndev** [gitbutlerapp/gitbutler#4061] [Q&A] [answered]
Is there a reason why the tree view was removed in #3416? Moving bigger changesets around is much easier in folder-based tree view.

**Comment by lextatic** [comment]
It seems the tree view was removed due to a redesign of how changes are displayed on lanes, but I urge you to consider bringing it back at some point. In my opinion, tree views are a much better way to understand and parse changes, especially in projects with many folders.

**Comment by PavelLaptev** [comment, marked as answer]
I agree. We're planning to bring back the tree view, likely with the new design update that's currently in progress.
We removed the tree view earlier to simplify the development of the file selection feature (using hotkeys like `Shift` and `Cmd`).

**Reply by ivan-toriya** [reply]
Is there any updates when the tree view will be back?

**Reply by PavelLaptev** [reply]
Sorry for the delayed response, @ivan-toriya. 
We'll most likely bring it back in V3. I’m planning to work on the design in a few weeks.

**Comment by PavelLaptev** [comment]
It's back :-)

<img width="623" height="498" alt="image" src="https://github.com/user-attachments/assets/6fe00967-4011-439e-aada-84a2684c0f8c" />
